paasta_tools/setup_marathon_job.py

```diff
@@ -242,6 +243,11 @@ def log_bounce_action(line, level='debug'):
 
     kill_given_tasks(client=client, task_ids=[task.id for task in tasks_to_kill], scale=True)
 
+    for task in bounce_lib.flatten_tasks(old_app_at_risk_tasks):
+        if task in tasks_to_kill:
+            hostname = task.host
+            reserve_all_resources([hostname])
```

Comment:
I'm glad this is only a small addition here 👍  I guess it could be pulled into `kill_given_tasks` which might mean the reservation happens sooner after a task is killed? However, I think we should try it like this first because it reads more cleanly.